build_library/build_image_util.sh

```diff
@@ -273,6 +273,58 @@ write_packages() {
     image_packages "$1" | sort > "$2"
 }
 
+get_metadata() {
+    local prefix="$1"
+    local pkg="$2"
+    local key="$3"
+    local path="${prefix}/var/db/pkg/${pkg%%:*}/${key}"
+    local val
+    if [[ -f "$path" ]]; then
+        val="$(< $path)"
+    else
+        # The package is not installed in $prefix or $key not exposed as file,
+        # so get the value from its ebuild
+        val=$(portageq-${BOARD} metadata "${BOARD_ROOT}" ebuild \
```

Comment:
Below you say that the portage wrapper is slow. Is this the command you're talking about?  You're invoking this command several times per package (one per key). If it's slow, it might be worth it to split this function in two to avoid calling the wrapper over and over.

Reply:
I moved this code. It already tries to read the value from a file if it exists but that doesn't work for all packages and it's not working for SRC_URI values.